_rules/document-has-instrument-to-main-ye5d6e.md

```diff
@@ -0,0 +1,688 @@
+
+## Expectation 1
+
+Within the test target, there exists at least one [keyboard actionable][] [instrument][] to move focus [at the start][] of the [main block of content][] of the [document][].
+
+## Expectation 2
+
+Within the test target, there exists at least one [instrument][] to move focus [at the start][] of the [main block of content][] of the [document][]; and this [instrument][] is [included in the accessibility tree][] and has an [accessible name][] that communicates that it skips to the [main block of content][].
+
+## Assumptions
+
+- This rule assumes that there is exactly one [main block of content][] inside each [HTML web page][].
+- This rule assumes that [Technique G1: Adding a link at the top of each page that goes directly to the main content area][tech g1], [Technique G123: Adding a link at the beginning of a block of repeated content to go to the end of the block][tech g123], and [Technique G124: Adding links at the top of the page to each area of the content][tech g124] require that the links can be [activated][activation] by use of keyboard, including being part of [sequential focus navigation][] (in order to be useful for keyboard users).
+- This rule assumes that there is at least one [block of repeated content][] before the [main block of content][], and therefore [Technique G123: Adding a link at the beginning of a block of repeated content to go to the end of the block][tech g123] will require a link to the [main block of content][] in order to skip this [block of repeated content][]. If there is no [block of repeated content][] before the [main block of content][], then it is possible to fail this rule but still pass [Technique G123: Adding a link at the beginning of a block of repeated content to go to the end of the block][tech g123].
+- This rule assumes that the language of each test target can be correctly determined (either programmatically or by analyzing the content), and sufficiently understood.
+
+## Accessibility Support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+While it is clear that a "skip link" is a valid way to satisfy [Success Criterion 2.4.1 Bypass blocks][sc241], it is less clear how "deep" in the page such a skip link could be. Notably, [Technique G124: Adding links at the top of the page to each area of the content][tech g124] is listing valid cases where it could be fairly "deep" if the page has many areas of the content. Rather than trying to fix an arbitrary value (e.g. "the skip link must be among the first 5 focusable elements"), or trying to figure out some condition on what precedes it, this rule only checks its existence. It is clear that if no "skip link" is provided, then another way to bypass blocks of repeated content must be found. However, it is possible to pass this rule and still fail [Success Criterion 2.4.1 Bypass blocks][sc241] if the skip link is too far away from the start of the page.
```

Comment:
I don't quite get the problem here. If the skiplink comes after repeating content, then that piece of repeating content can not be bypassed, and so that skiplink could not be used to satisfy the SC.

Reply:
There has been a lot of discussion that lead to the rule as it is and this note. There are essentially 4 cases that sum up that:
* https://github.com/act-rules/act-rules.github.io/blob/dc595d82a9f8a4378baa8800ae7ed1dae7d09849/_rules/bypass-blocks-cf77f2.md#passed-example-11 has a skip link as its second link. Reviewers agreed that it pass 2.4.1. @adilsofficial was mentioning that he usually accept skip links up to 3 tab-stops away from the start. Note that the first link here is "random" and has nothing to do with the page.
* https://github.com/act-rules/act-rules.github.io/blob/dc595d82a9f8a4378baa8800ae7ed1dae7d09849/_rules/document-has-instrument-to-main-ye5d6e.md#passed-example-2 has a link to main which is the 3rd tab stop. This one clearly passes G124, hence satisfies 2.4.1 It is easy to build on this example to have a case that passes G124 but with a skip link 10 (or more) tab-stops away from the start (at any arbitrary depth, actually).
* https://github.com/act-rules/act-rules.github.io/blob/dc595d82a9f8a4378baa8800ae7ed1dae7d09849/_rules/first-focusable-element-is-link-to-main-8a213c.md#passed-example-5 In this example, the skip link is **after** the repeated content, but since there is no tab-stop in the repeated content, the skip link is still the first.
* https://github.com/act-rules/act-rules.github.io/blob/dc595d82a9f8a4378baa8800ae7ed1dae7d09849/_rules/document-has-instrument-to-main-ye5d6e.md#passed-example-3 Here, the "true" skip link (the one that goes to main) is inside the second repeated block, after the first one, and possibly many links inside it. This passes G123 so satisfies 2.4.1

These examples show that we can't just force the location of the skip link based on its "depth", on previous links, or on repeated content. We might be able to formulate something based on all that, but it might be too complex.
In the end, we agreed with the reviewer than we are safe in asking a "link to main" somewhere in the page. We can't safely enforce conditions of where in the page it should be without risking false positives, but we can definitely fail if there is no link to main at all. This is also why "first focusable is link to main" become an independent best practice rule.

This note here tries to explain this.